ncrypt/crypt_gpgme.c

```diff
@@ -2518,11 +2518,16 @@ static int pgp_gpgme_extract_keys(gpgme_data_t keydata, FILE **fp)
         fprintf(*fp, "sub %5.5s %u/%8s %s\n", gpgme_pubkey_algo_name(subkey->pubkey_algo),
                 subkey->length, shortid, date);
       }
-      else
+      else if (uid)
       {
         fprintf(*fp, "pub %5.5s %u/%8s %s %s\n", gpgme_pubkey_algo_name(subkey->pubkey_algo),
                 subkey->length, shortid, date, uid->uid);
       }
+      else
+      {
+        fprintf(*fp, "pub %5.5s %u/%8s %s\n", gpgme_pubkey_algo_name(subkey->pubkey_algo),
+                subkey->length, shortid, date);
+      }
```

Comment:
if you can live with an extra space at the end of the line:

```suggestion
      else
      {
        fprintf(*fp, "pub %5.5s %u/%8s %s %s\n", gpgme_pubkey_algo_name(subkey->pubkey_algo),
                subkey->length, shortid, date, (uid ? uid->uid : ""));
      }
```

Reply:
I thought about that...

> if you can live with an extra space at the end of the line:

The perfectionist in me shuddered at the thought!
The text is displayed to the user, but not used anywhere 'critical', so I can live with a non-visible space.